### PR TITLE
Simple audio torch version

### PR DESCRIPTION
**Pull request by Athospd**
Hello! I wrote a post that is a translation of a previous post by Daniel, from tensorflow/keras to torch/torchaudio to illustrate the usage of torchaudio package along with torch itself. I would like to offer it as content for the blog.

Known issues are: 
- poor English writing.
- performance/time taken to fit is somewhat worse than the original post. Just for the record, I once managed to reach 93% val_acc, but for some reason, I couldn't reproduce it anymore! =( If one happen to have access to a good machine, I'd encourage to rebuild de Rmd. It took me about 25 minutes.

Best regards!

### CODE REVIEW

**Comment by skeydan** [comment]
Hi @Athospd, many thanks for the PR!!
I'll take a look as soon as possible.

File: _posts/2021-01-29-simple-audio-classification-with-torch/simple-audio-classification-with-torch.Rmd [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+set_progress_bar <- function(total) {
+  progress_bar$new(
+    total = total, clear = FALSE, width = 70,
+    format = ":current/:total [:bar] - :elapsed - loss: :loss - acc: :acc"
+  )
+}
+```
+
+{torchaudio} comes with 3 loaders, av_loader, tuner_loader, and audiofile_loader - more to come. The `set_audio_backend()` is used to set one of them as the audio loader. Their performances differs from audio format (mp3 or wav). There is no perfect world yet: tuner_loader is the best for mp3, audiofile_loader is the best for wav, but both do not have de option of partial load a sample from audio file without bringing all the data to memory first.
```
Comment (skeydan) [inline _posts/2021-01-29-simple-audio-classification-with-torch/simple-audio-classification-with-torch.Rmd:288]:
```suggestion
{torchaudio} comes with 3 loaders, `av_loader`, `tuner_loader`, and `audiofile_loader`- more to come.  `set_audio_backend()` is used to set one of them as the audio loader. Their performances differ based on audio format (mp3 or wav). There is no perfect world yet: `tuner_loader`is best for mp3, `audiofile_loader` is  best for wav, but neither of them has the option of partially loading a sample from an audio file without bringing all the data into memory first.
```

File: _posts/2021-01-29-simple-audio-classification-with-torch/simple-audio-classification-with-torch.Rmd [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  x %>% select(class, pred_class),
+  freq = x$n,
+  col = ifelse(x$correct, "lightblue", "red"),
+  border = ifelse(x$correct, "lightblue", "red"),
+  alpha = 0.6,
+  hide = x$n < 20
+)
+```
+
+Model accuracy is `r scales::percent(val_acc, accuracy = 0.01)`, similar to tensorflow version from the original post. Likewise, all conclusions from original post also holds.
```
Comment (skeydan) [inline _posts/2021-01-29-simple-audio-classification-with-torch/simple-audio-classification-with-torch.Rmd:357]:
```suggestion
Model accuracy is `r scales::percent(val_acc, accuracy = 0.01)`, similar to the tensorflow version from the original post. Likewise, all conclusions from original post also hold.
```

**Review by skeydan — COMMENTED** [review]
Hi, 
thanks again for the PR, and first of all, thanks for porting torchaudio!
I've left a few suggestions below - whenever you agree, you can just commit them, or even better, add them to a batch that you can commit once you're ready.

A few other points:

- We've changed how the blog is built. This means that unlike before, no generated files should be under version control. In your case, this should really just leave the .Rmd and the preview image. Can you please remove all other files?

- Right now, the deploy preview fails. This might be because of the generated files. Let's see if we can get a working preview once everything besides the Rmd and the image are removed.

- If I'm correct, you're evaluating the Rmd "live", right? This is something we want to avoid, to reduce build times. Instead, we add all output statically (copy-paste), and have a chunk that says

**Comment by skeydan** [comment]
Oh, one thing I should say ... We've noticed that Github is super quirky in that it hides unresolved suggestions. One has to periodically click "show hidden conversations" until one's finally seen them all :-(

File: _posts/2021-01-29-simple-audio-classification-with-torch/simple-audio-classification-with-torch.Rmd [hunk 3]
```diff
@@ -0,0 +1,358 @@
+---
+title: "Simple audio classification with torch"
+description: > 
+  This article attempts to translate the [Daniel Falbel's](https://github.com/dfalbel) ['Simple Audio Classification'](https://blogs.rstudio.com/ai/posts/2018-06-06-simple-audio-classification-keras/) article from `{tensorflow}+{keras}` to `{torch}+{torchaudio}`.
+author:
+  - name: Athos Damiani
+    affiliation: Curso-R
+    affiliation_url: https://curso-r.com/
+slug: athossimpleaudioclassification
+date: 01-29-2021
```
[the remark this answers is not in the document]
Comment (Athospd) [reply _posts/2021-01-29-simple-audio-classification-with-torch/simple-audio-classification-with-torch.Rmd]: OK! I set 'date:' and folder name to 04-02-2021.

**Comment by Athospd** [comment]
Thank you very much, @skeydan !!

About the to-do list:
- [x] Commit suggested text/grammar + typo fix
- [x] update dates from Rmd and folder
- [x] delete generated files
- [ ] deploy preview fails - **I suspect this issue is due to torch and torchaudio nighly versions dependency =( should I rollback to depend solely on CRAN versions?**
- [x] copy-paste output chunkwise + `knitr::opts_chunk$set(echo = TRUE, eval = FALSE)`
- [x] remove `{` and `}` from paragraphs.

**Comment by skeydan** [comment]
Wonderful, thank you! I think the failure is due to this line, where R code still gets evaluated:

**Comment by Athospd** [comment]
> ```
> Model accuracy is `r scales::percent(val_acc, accuracy = 0.01)`, similar to the tensorflow version from the original post. Likewise, all conclusions from original post also hold.
> ```

Fixed!

**Comment by skeydan** [comment]
Hi Athos, thank you!

Preview still fails for unrelated reasons, and the current workaround (we'll be worker on a better one) involves me checking out the file locally and then, creating a new PR. 

While I'm at it, would you mind if I make remaining edits myself? I see there are some missing spaces and the like (this stuff is hard to see when reviewing the raw Rmd).

After I've created the new PR, I'll put a link here with the preview URL.

BTW: I was going to ask "why did you remove the nice preview image?" - but then I saw that the file has `.jpeg` while in the header it says `.jpg` :-) ... so it's all good ;-)))

**Comment by skeydan** [comment]
here's the preview: https://601ac5f14e9bf7008f9ca186--hardcore-kirch-782335.netlify.app/posts/2021-02-04-simple-audio-classification-with-torch/

**Comment by Athospd** [comment]
Awesome! Just for the record, I took the preview image from [Unsplash](https://unsplash.com/photos/ASKeuOZqhYU), so we are safe =)

And please feel free to edit anything! 

Let me take the opportunity to thank you for your amazing work sharing your knowledge on this blog. Your posts are already in the .bib file of my master's dissertation.

**Comment by skeydan** [comment]
Thank you so much! That's very kind of you :-)

And thank you very much for porting torchaudio!
As you can probably guess (from _my_ follow-up to Daniel's trend-setting post ;-))) I'm also very interested in audio processing, and I've had torchaudio on my list for quite some time --- but what could be better than a post by the creator themselves? :-)

**Comment by Athospd** [comment]
My English speaking skills do not allow me to express the way I want, but, in summary: 1) I'm very interested in audio processing as well, I'm studying a lot! 2) Please share your "adventures in audio world" as much as you can, your convLSTM was inspiring. 3) torchaudio has lots of missing parts yet, especially nn_modules and datasets, so please open an issue if you miss something!

**Pull request merged**